### PR TITLE
[Docs] Add painless context details for bucket_script

### PR DESCRIPTION
**Pull request by polyfractal**
Fleshes out the details of the `bucket_script` context, adds an example and corresponding doc test.  I decided not to add a test to `ContextExampleTests` because the script is trivial, and we can test it through the usual doc testing mechanism.

Related to #34829

### CODE REVIEW

**Review by jdconrad — COMMENTED** [review]
@polyfractal Thank you for this! This looks good from the side to me.  I would like @debadair or another writer to take a look for the docs side.

File: docs/painless/painless-contexts.asciidoc [hunk 1]
```diff
@@ -44,7 +44,7 @@ specialized code may define new ways to use a Painless script.
                                     | {ref}/search-aggregations-metrics-scripted-metric-aggregation.html[Elasticsearch Documentation]
 | Metric aggregation reduce         | <<painless-metric-agg-reduce-context, Painless Documentation>>
                                     | {ref}/search-aggregations-metrics-scripted-metric-aggregation.html[Elasticsearch Documentation]
-| Bucket aggregation                | <<painless-bucket-agg-context, Painless Documentation>>
+| Bucket Script aggregation         | <<painless-bucket-script-agg-context, Painless Documentation>>
                                     | {ref}/search-aggregations-pipeline-bucket-script-aggregation.html[Elasticsearch Documentation]
```
Comment (debadair) [inline docs/painless/painless-contexts.asciidoc]: I would lowercase script. In general, we don't use init caps in names of things

File: docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+To run this example, first follow the steps in <<painless-context-examples, context examples>>.
+
+The painless context in a `bucket_script` aggregation provides a `params` map.  This map contains both
+user-specified custom values, as well as the values from other aggregations specified in the `buckets_path`
+property.
+
+For our example, we are going to take the values from a `min` and `max` aggregation, subtract them to find
+the difference, then add in a "base cost" from a user-specified value.  The script looks like this:
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc]: In general, I recommend avoiding the use of "we" or "our". In this case, I'd just say:

This example takes the values from a `min` and `max` aggregation, calculates the difference, and adds the user-specified `base_cost` to the result.

File: docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+[source,Painless]
+--------------------------------------------------
+(params.max - params.min) + params.base_cost
+--------------------------------------------------
+
+Note how all the values are being pulled from the `params` map.  In context, the aggregation looks like this:
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc]: I'd pare this first sentence down to: 

Note that the values are extracted from the `params` map.

File: docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:seats]
+<1> the `buckets_path` points to two aggregations (`min_cost`, `max_cost`) and adds `min`/`max` variables
+to the `params` map
+<2> the user-specified `base_cost` is added in the script's `params` section, which is also added to the
+`params` variable for use in the script.
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc]: I think we generally (but likely not consistently) init-cap callouts whether or not they are complete sentences.

In <2>, I'd refer to it as the `params` map for consistency. I think I'd also drop "for use in the script".

**Review by debadair — CHANGES_REQUESTED** [review]
Suggested some minor edits.

**Comment by polyfractal** [comment]
Thanks @debadair.  Pushed some tweaks per your suggestions :)

**Review by debadair — APPROVED** [review]

**Pull request merged**